src/main/webapp/static/js/controller/pokemon_controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module('myApp').controller('PokemonController', ['$scope', 'PokemonService', function ($scope, PokemonService) {
+    var self = this;
+    self.pokemon = { id: null, name: '', attack: '', defense: '' };
```

Comment:
note that these object members are dynamically declared, and may not be accurate with your java model.  Default to just declaring empty objects like,  self.pokemon = {};  and use your "Network" tab to inspect the JSON models.